### ISSUE TITLE
getScope should retrieve module scope instead of global scope for ES6 modules

### ISSUE THREAD

**Issue by voideanvalue** [eslint/eslint#3700]
escope separates `global` scope from `module` scope for ES6 modules by [nesting a `module` scope within the `global` scope](https://github.com/estools/escope/blob/45cf8512857ac5663866d2a08d1d8fb28f78366f/src/referencer.js#L496-l498). `getScope` currently [`acquire`s](https://github.com/eslint/eslint/blob/fe3a047b6b29e97597c8e8f9882f168c17e15a08/lib/eslint.js#L967) [the outermost scope](https://github.com/estools/escope/blob/6a45f68c9aa3937dcb37fe177932683b84892b9a/src/scope-manager.js#L95-L102) from escope. This leads to the `global` scope being returned when `getScope` is called in context of a top level statement instead of the `module` scope.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
This is the correct behavior. For Program, the associated scope is the global scope.

**Comment by nzakas** [comment]
Although, reading this again, I'm not even sure that's what you mean. It would be helpful for you to provide a use case and sample code.

Note: it's unlikely we will make a change here as it would likely be a breaking change, but I'd still like to understand what it is you're trying to do.

**Comment by voideanvalue** [comment]
@nzakas

Consider
```js
var foo = {};
foo.bar = 1;
```

`getScope` in context of the `AssignmentExpression` on line 2 would return the `global` scope which doesn't contain the variable `foo` (the `module` scope does). Is that intended?

**Comment by nzakas** [comment]
Ah, so you're not accessing the scope from Program, but rather from a deeper node? Yeah, that looks like a bug. I'm guessing it also affects when `globalReturn` is true, as that also adds an additional scope.

**Comment by voideanvalue** [comment]
Exactly. Do you think the approach I took in #3701 is acceptable? I think always picking the inner most scope makes sense because it contains everything from the outer scope(s) anyway. It also seems cleaner than monkey patching for specific cases like this one. Plus, the fact that all unit tests pass with the change suggests that in most cases there is only one scope for a node (which would've been my guess). Not sure if you'd still consider it a breaking change or not though.

And good point about `globalReturn`. I believe #3701 should also fix that. I can add a unit test for that too, if you'd like.
